README.md
## X.Da小项目:bow:

[![Build Status](https://travis-ci.org/thianda/phpweb.svg?branch=master)](https://travis-ci.org/thianda/phpweb)

基于ThinkPHP 5.0.x开发

更新log可查看 [CHANGELOG](CHANGELOG.md)

目前模块有：

| 模块名       | 功能描述                                     |
| --------- | ---------------------------------------- |
| docs      | 开发文档库( *beta*) :fire:                    |
| checkME60 | 城域网BAS设备健康度检查(**done**) :heavy_check_mark: |
| common    | 公共库                                      |
| esserver  | excel服务器辅助(**done**) :heavy_check_mark:  |
| index     | 首页                                       |
| kxemas    | 客响维材管理系统(*shutdown)  :no_entry_sign:     |
| othermode | template模板                               |
| trouble   | 办公终端报修流程(*pause) :wrench:                |
| zx_apply  | 专线业务开通辅助(*alpha*) :fire:                 |

本地 clone 后需要在本目录下执行一次 `composer install`

composer 的安装及使用请查看： [composer 中文网](http://www.phpcomposer.com/ )

> 另：`public/static/`内文件已打包，不保存在repo中。详见[release](https://github.com/thianda/phpweb/releases)。
>
> `docs`需搭环境（*jekyll*）在`docs/`内执行`x.da-build-sh.sh`生成静态文档。

### 生成docs步骤

以windows系统为例，安装git-for-windows后，使用 git-bash，切换到`/docs`目录

 1. 安装 rubyinstaller

   [官网下载](https://rubyinstaller.org/downloads/)安装

 2. 安装 jekyll

   ```sh
   gem install jekyll
   ```
   安装出错请参考[这里](https://thianda.github.io/notes/create-blog-by-using-jekyll.html#%E5%AE%89%E8%A3%85jekyll)。

 3. 安装 bundler

   ```sh
   gem install bundler
   ```

4. 安装依赖

  ```sh
  bundle install
  ```

5. 生成静态文档到`/public/phpweb`

  ```sh
  ./x.da-build-sh.sh
  ```


CHANGELOG.md
## 更新日志

### v0.6.0(2018-07-13)

[zx_apply]模块更新

- 实现ONU业务申请时无需`A端基站`，先预分配ip不分配vlan，待做ONU上线数据时再选择`A端基站`并分配`vlan`。
- apply.html申请、query.html修改申请、todo代办页面同步修改表单动态逻辑
- 修改`Vlantables::check()`逻辑，优化存在`vlan`但不存在`aStation`时的结果

#### v0.5.5(2018-07-13)

[zx_apply]模块更新

- 调整：申请时若为ONU业务则不需要填写`A端基站`，此规则在`修改申请`时亦生效

- 修复`manage/query.html`。在此之前，批量删除时前端显示与实际不符
- 完善vlan删除与修改功能：修改状态`[0]申请`，到待办里直接修改提交即可，若不填vlan即为删除
- 修复其他若干bug

#### v0.5.1(2018-07-12)

[zx_apply]模块代码格式化

- 格式化`zx_apply`模块的代码（php、html），遵循PSR-2
- 未知原因导致部分代码丢失，已补回

### v0.5.0(2018-07-11)

[zx_apply]模块部分重构

- 前端部分重构
- 修复申请填表单的bug
- 需进一步调整

#### v0.4.18(2018-07-11)

[zx_apply]模块更新

- 更新资管导入模板
- 优化`XLSX`组件的加载逻辑
- 计划将`导出excel操作`移至前端，加速导出速度，兼容性更强
- 修复细节bug

#### v0.4.16(2018-07-05)

[zx_apply]模块更新

- 修改`index/query.html`中`afterChange`事件的bug
- 新增ip列表生成：`tool/swscript.html`

#### V0.4.13(2018-06-28)

[zx_apply]模块更新

- `manage/query.html`新增批量已完成功能，批量选取设置状态为[4]:已完成
- 导出资管模板数据时提示批量修改状态为[2]:已录资管等待IP备案

TODO：`更多操作`页面新功能：统计空闲ip

> 需基于专线类型，细节有待研究

#### V0.4.12(2018-06-26)

[zx_apply]模块修复若干bug

- 完善细节功能，修复大量bug
- 支持按照`专线类型`进行`全局搜索`以及`基本信息查询`

#### v0.4.9(2018-06-12)

- 用户可右键选择`修改申请`，修改个别字段信息后重新提交
- 用户申请提交前自动清除每个字段里的空格并给出提示

#### v0.4.8(2018-06-06)

[zx_apply]模块bug修复

- 累计优化多处UI布局
- `todo`页提示同客户名的台账信息
- `apply`页提交前过滤空格符
- 模板导出时文件名的时间字段放后面

#### v0.4.6(2018-05-29)

[zx_apply]模块更新

- 视图显示优化、安全事件逻辑优化
- 显示和导出全量数据可基于专线类别筛选，申请与待办暂未实现
- 查询、导出操作均记录log

#### V0.4.5(2018-05-24)

[zx_apply]模块bug修复

- 修复ip分配时的小bug
- 调整查询页面排序为`status,create_time desc`
- 更新资管系统导出模板
- 调整各种模板导出时文件名的命名规则

#### v0.4.4(2018-05-18)

[zx_apply]模块大量更新。

- 待办页`vlan`字段不再是必填
- 待办页“自动预分配”结果显示由`alert`改为`window`
- 查询页显示详细信息布局优化
- 查询页取消显示备注列
- 调整待办通知的邮件内容
- 申请人查询页可查询历史数据的基本信息（去除客户联系方式等敏感信息，并限制查询次数）

#### v0.4.3(2018-05-16)

- [zx_apply]模块预分配ip后给申请人自动发送邮件提醒（手动判断是否发送）

#### v0.4.2(2018-05-11)

#### v0.4.1(2018-05-04)

[zx_apply]模块开始公测使用

- 添加`jsSheet`组件，用于zx_apply前端导出全量台账的xlsx文件
- 敏感配置信息独立存储

### v0.4.0(2018-04-02)

[zx_apply]模块大量更新。

- `Infotables`增加 ip掩码和ipB掩码，互联网ip掩码默认32，其他默认30，所有ipB掩码默认29
- 取消`Iptables`数据表，保留其model类及方法
- 定位为辅助平台，提供有限的台账维护功能。主要功能为限制数据录入的规范性。导出为各项工作所需的数据。包含但不限于：
  - 台账格式
  - 资管流程模板格式
  - 做数据脚本
  - ip备案模板格式（2种）
- 管理员查询页面可右键调用数据制作脚本生成、资管系统录入模板、集团IP备案导入模板、工信部IP备案导入模板功能。
- 更新`handsontable`组件到v1.18.1/v0.38.1

### v0.3.2(2018-01-31)

- 更新`dhtmlx`组件到5.1.0
- 修复esserver的bug
- 添加系统log记录功能
- 添加zx_apply模块的登陆控制

#### v0.2.2.0108(2018-01-08)

- 更新`/docs/_posts/2017-12-21-handsontable-usage-notes.md`内容

> 总结的自己脑袋都乱了。官网这个[docs](https://docs.handsontable.com)真的一点也不友好。

- 更新zx_apply核心代码

#### v0.2.2.0107(2018-01-07)

- 更新`/docs/_posts/2017-12-21-handsontable-usage-notes.md`内容
- 添加`handsontable`作为grid组件。与`dhtmlXGrid`相比的优点：
  - 有配置默认值，上手更简单
  - UI更接近Excel，功能也更近似Excel
  - 可全局搜索，性能良好

#### v0.2.2(2017-12-22)

- 更新docs入口为项目名，可在github pages自动发布
- 添加静态资源handsontable
- 更新docs开发文档样式，修复几处bug

### v0.2.0(2017-12-18)

- 添加docs，记录开发文档
- 大量微调整

### v0.1.0(2017-11-16)

开始托管于github，整合多个小项目（checkME60、esserver、kxeams、trouble、zx_apply）为本项目的子模块。

- initial release

application/zx_apply/config.php
<?php
return [ 
		'version' => "V0.6.0-zx",
		'db_esserver' => [ 
				'type' => 'Sqlsrv',
				// 服务器地址
				'hostname' => '223.100.98.60',
				// 数据库名
				'database' => 'ESapp1',
				// 用户名
				'username' => 'sa',
				// 密码
				'password' => 'Zz123456',
				// 端口
				'hostport' => '1433',
				// 数据库编码默认采用utf8
				'charset' => 'gbk2312',
				// 数据库表前缀
				'prefix' => 'dbo.ES_' 
		
		],
		// 'aStation' => '{"柴河局09":"CHJ-09","西丰新局":"XF-10","昌图新局":"CT-11","昌图老局":"CT-12","西丰老局":"XF-13","开原老局":"KY-14","铁岭县":"TLX-15","清河局":"QH-16","新台子":"XTZ-17","调兵山新局":"DBS-18","调兵山老局":"DBS-19","开原新局":"KY-20","柴河局-华为":"CHJ-21","柴河局-中兴":"CHJ-22","银州区":"YZL-23"}',
		'aStation' => [ 
				"请选择" => null,
				"柴河局09" => "CHJ-09",
				"西丰新局" => "XF-10",
				"昌图新局" => "CT-11",
				"昌图老局" => "CT-12",
				"西丰老局" => "XF-13",
				"开原老局" => "KY-14",
				"铁岭县" => "TLX-15",
				"清河局" => "QH-16",
				"新台子" => "XTZ-17",
				"调兵山新局" => "DBS-18",
				"调兵山老局" => "DBS-19",
				"开原新局" => "KY-20",
				"柴河局-华为" => "CHJ-21",
				"柴河局-中兴" => "CHJ-22",
				"银州区" => "YZL-23",
				"柴河局" => null
		],
		'device9312' => '{"CHJ-09":{"up_port_yz":"Eth-Trunk12","up_port_ch":"Eth-Trunk9","port_hw":"Eth-Trunk36","port_zte":"Eth-Trunk26","bas02_down_port":"Eth-Trunk9","bas01_down_port":"Eth-Trunk9","rbp_name":"ch-yz-sw09-chj9312-jk"},"XF-10":{"up_port_yz":"Eth-Trunk114","up_port_ch":"Eth-Trunk14","port_hw":"Eth-Trunk2","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk51","bas01_down_port":"Eth-Trunk3","rbp_name":"ch-yz-sw10-xf9312-pppoe"},"CT-11":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","port_hw":"Eth-Trunk6","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk41","bas01_down_port":"Eth-Trunk41","rbp_name":"ch-yz-sw11-ct9312-pppoe"},"CT-12":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk42","port_zte":"Eth-Trunk43","bas02_down_port":"Eth-Trunk42","bas01_down_port":"Eth-Trunk42","rbp_name":"ch-yz-sw12-ct9312"},"XF-13":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_zte":"Eth-Trunk18","bas02_down_port":"Eth-Trunk46","bas01_down_port":"Eth-Trunk46","rbp_name":"ch-yz-sw13-xf9312"},"KY-14":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","port_hw":"Eth-Trunk23","port_zte":"Eth-Trunk24","bas02_down_port":"Eth-Trunk43","bas01_down_port":"Eth-Trunk43","rbp_name":"ch-yz-sw14-ky9312-pppoe"},"TLX-15":{"up_port_yz":"Eth-Trunk114","up_port_ch":"Eth-Trunk14","port_hw":"Eth-Trunk2","bas02_down_port":"Eth-Trunk45","bas01_down_port":"Eth-Trunk45","rbp_name":"ch-yz-sw15-tlx9312-jk"},"QH-16":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","port_hw":"Eth-Trunk13","port_zte":"Eth-Trunk7","bas02_down_port":"Eth-Trunk6","bas01_down_port":"Eth-Trunk6","rbp_name":"ch-yz-sw16-qh9312-jk"},"XTZ-17":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","bas02_down_port":"Eth-Trunk47","bas01_down_port":"Eth-Trunk47","rbp_name":"ch-yz-sw17-xtz9312-jk"},"DBS-18":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk3","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk53","bas01_down_port":"Eth-Trunk54","rbp_name":"ch-yz-sw18-dbs9312"},"DBS-19":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","bas02_down_port":"Eth-Trunk49","bas01_down_port":"Eth-Trunk49","rbp_name":"ch-yz-sw19-dbs9312"},"KY-20":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","port_hw":"Eth-Trunk27","port_zte":"Eth-Trunk26","bas02_down_port":"Eth-Trunk50","bas01_down_port":"Eth-Trunk50","rbp_name":"ch-yz-sw20-ky9312-pppoe"},"CHJ-21":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk41","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk39","bas01_down_port":"Eth-Trunk39","rbp_name":"ch-yz-sw21-chj9312-pppoe"},"CHJ-22":{"up_port_yz":"Eth-Trunk1","up_port_ch":"Eth-Trunk0","port_zte":"Eth-Trunk17","bas02_down_port":"Eth-Trunk59","bas01_down_port":"Eth-Trunk59","rbp_name":"ch-yz-sw22-chj9312-pppoe"},"YZL-23":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","bas02_down_port":"Eth-Trunk19","bas01_down_port":"Eth-Trunk19","rbp_name":"ch-yz-sw23-yzl9312-pppoe"}}',
		// 'device9312' => '{"柴河-华为":{"up_port_yz":"Eth-Trunk12","up_port_ch":"Eth-Trunk9","port_hw":"Eth-Trunk36","port_zte":"Eth-Trunk26","bas02_down_port":"Eth-Trunk9","bas01_down_port":"Eth-Trunk9","rbp_name":"ch-yz-sw09-chj9312-jk"},"柴河-中兴":{"up_port_yz":"Eth-Trunk114","up_port_ch":"Eth-Trunk14","port_hw":"Eth-Trunk2","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk51","bas01_down_port":"Eth-Trunk3","rbp_name":"ch-yz-sw10-xf9312-pppoe"},"西丰新局":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","port_hw":"Eth-Trunk6","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk41","bas01_down_port":"Eth-Trunk41","rbp_name":"ch-yz-sw11-ct9312-pppoe"},"西丰老局":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","port_hw":"Eth-Trunk23","port_zte":"Eth-Trunk24","bas02_down_port":"Eth-Trunk43","bas01_down_port":"Eth-Trunk43","rbp_name":"ch-yz-sw14-ky9312-pppoe"},"昌图新局":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk42","port_zte":"Eth-Trunk43","bas02_down_port":"Eth-Trunk42","bas01_down_port":"Eth-Trunk42","rbp_name":"ch-yz-sw12-ct9312"},"昌图老局":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_zte":"Eth-Trunk18","bas02_down_port":"Eth-Trunk46","bas01_down_port":"Eth-Trunk46","rbp_name":"ch-yz-sw13-xf9312"},"开原老局":{"up_port_yz":"Eth-Trunk114","up_port_ch":"Eth-Trunk14","port_hw":"Eth-Trunk2","bas02_down_port":"Eth-Trunk45","bas01_down_port":"Eth-Trunk45","rbp_name":"ch-yz-sw15-tlx9312-jk"},"开原新局":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","port_hw":"Eth-Trunk27","port_zte":"Eth-Trunk26","bas02_down_port":"Eth-Trunk50","bas01_down_port":"Eth-Trunk50","rbp_name":"ch-yz-sw20-ky9312-pppoe"},"清河":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","bas02_down_port":"Eth-Trunk47","bas01_down_port":"Eth-Trunk47","rbp_name":"ch-yz-sw17-xtz9312-jk"},"调兵山新局":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk3","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk53","bas01_down_port":"Eth-Trunk54","rbp_name":"ch-yz-sw18-dbs9312"},"调兵山_老":{"up_port_yz":"Eth-Trunk112","up_port_ch":"Eth-Trunk12","bas02_down_port":"Eth-Trunk49","bas01_down_port":"Eth-Trunk49","rbp_name":"ch-yz-sw19-dbs9312"},"铁岭县":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","port_hw":"Eth-Trunk13","port_zte":"Eth-Trunk7","bas02_down_port":"Eth-Trunk6","bas01_down_port":"Eth-Trunk6","rbp_name":"ch-yz-sw16-qh9312-jk"},"银州区":{"up_port_yz":"Eth-Trunk113","up_port_ch":"Eth-Trunk13","port_hw":"Eth-Trunk41","port_zte":"Eth-Trunk5","bas02_down_port":"Eth-Trunk39","bas01_down_port":"Eth-Trunk39","rbp_name":"ch-yz-sw21-chj9312-pppoe"},"新台子":{"up_port_yz":"Eth-Trunk1","up_port_ch":"Eth-Trunk0","port_zte":"Eth-Trunk17","bas02_down_port":"Eth-Trunk59","bas01_down_port":"Eth-Trunk59","rbp_name":"ch-yz-sw22-chj9312-pppoe"},"柴河09":{"up_port_yz":"Eth-Trunk2","up_port_ch":"Eth-Trunk1","bas02_down_port":"Eth-Trunk19","bas01_down_port":"Eth-Trunk19","rbp_name":"ch-yz-sw23-yzl9312-pppoe"}}'
		// 额外信息，用于IP备案
		'extraInfo' => [ 
				"单位性质" => "unitProperty",
				"单位分类" => "unitCategory",
				"行业分类" => "industryCategory",
				"使用单位证件类型" => "credential",
				"使用单位证件号" => "credentialnum",
				"所在省" => "province",
				"所在市" => "city",
				"所在县区" => "county",
				"应用服务类型" => "appServType" 
		],
		// 登陆后跳转到manage模块的账号
		'manageEmails' => [ 
				"yuxianda.tl",
				"buyu.tl",
				"liuyutl.tl"
		],
		
		// 不再使用（直接在index/apply.html里定义）
		// 单位性质
		'unitProperty' => [ 
				1 => '军队',
				2 => '政府机关',
				3 => '事业单位',
				4 => '企业',
				5 => '个人',
				6 => '社会团体',
				7 => '国瑞',
				8 => '个体工商户',
				9 => '民办非企业',
				10 => '基金会',
				11 => '律师事务所',
				12 => '外国文化中心',
				99 => '未知' 
		],
		// 单位分类
		'unitCategory' => [ 
				1 => 'ISP',
				2 => 'IDC',
				3 => 'ISP和IDC',
				4 => '公益性互联网络单位',
				5 => '其他',
				11 => '电信业务经营者',
				12 => '公益性互联网络单位',
				14 => '管局单位',
				15 => '前置单位',
				999 => '未知' 
		],
		// 行业分类
		'industryCategory' => [ 
				1 => '农、林、牧、渔业',
				2 => '采矿业',
				3 => '制造业',
				4 => '电力、燃气及水的生产和供应业',
				5 => '建筑业',
				6 => '交通运输、仓储和邮政业',
				7 => '信息传输、计算机服务和软件业',
				8 => '批发和零售业',
				9 => '住宿和餐饮业',
				10 => '金融业',
				11 => '房地产业',
				12 => '租赁和商务服务业',
				13 => '科学研究、技术服务和地质勘查业',
				14 => '水利、环境和公共设施管理业',
				15 => '居民服务和其他服务业',
				16 => '教育',
				17 => '卫生、社会保障和社会福利业',
				18 => '文化、体育和娱乐业',
				19 => '公共管理与社会组织',
				20 => '国际组织',
				21 => 'example',
				999 => '未知' 
		],
		// 使用单位证件类型
		'credential' => [ 
				1 => '工商营业执照',
				2 => '身份证',
				3 => '组织机构代码证书',
				4 => '事业法人证书',
				5 => '军队代号',
				6 => '社团法人证书',
				7 => '护照',
				8 => '军官证',
				9 => '组织机构代码证书',
				10 => '组织机构代码证书' 
		],
		'province' => [ 
				210000 => '辽宁省' 
		],
		'city' => [ 
				211200 => '铁岭市' 
		],
		'county' => [ 
				211201 => '市辖区',
				211202 => '银州区',
				211204 => '清河区',
				211221 => '铁岭县',
				211223 => '西丰县',
				211224 => '昌图县',
				211281 => '调兵山市',
				211282 => '开原市' 
		],
		'appServType' => [ 
				1 => '企事业单位专线接入',
				2 => '个人专线接入',
				3 => '固定宽带接入',
				4 => '移动网接入',
				5 => '网络设备',
				6 => '网站专线接入',
				7 => 'IDC服务类',
				8 => '物联网' 
		],
		'colWidth' => [ 
				0 => 98,
				1 => 96,
				2 => 65,
				3 => 50,
				4 => 65,
				5 => 100,
				6 => 240,
				7 => 220,
				8 => 450,
				9 => 50,
				10 => 116,
				11 => 116,
				12 => 107,
				13 => 107,
				14 => 164,
				15 => 93,
				16 => 96,
				17 => 164,
				18 => 200,
				19 => 75,
				20 => 200,
				21 => 60,
				22 => 107,
				23 => 164,
				24 => 98,
				25 => 98,
				26 => 50,
				27 => 80,
				28 => 0,
				29 => 76,
				30 => 76,
				31 => 76,
				32 => 132,
				33 => 107,
				34 => 62,
				35 => 62,
				36 => 76,
				37 => 104 
		] 
];